spec/index.md

```diff
@@ -633,8 +641,64 @@ Calls must stay in `replied` or `rejected` long enough for polling users to catc
 
 When asking the IC about the state or call of a request, the user uses the request id (see [Request ids](#request-id)) to read the request status (see [Request status](#state-tree-request-status)) from the state tree (see [Request: Read state](#http-read-state)).
 
+#### Synchronous canister calling {#http-sync-call-overview}
+
+A synchronous update call, also known as a "call and await", is a type of update call where the replica will attempt to respond with a certificate of the state of the call to the HTTPS request. If the returned certificate indicates that the update call is in a terminal state (`replied`, `rejected`, or `done`), then the user __does not need to poll__ (using [`read_state`](#http-read-state) requests) to determine the result of the call. A terminal state means the call has completed its execution.
+
+The synchronous call endpoint is useful for users as it removes the networking overhead of polling the IC to determine the status of their call.
+
+The replica will maintain the HTTPS connection for the request and will respond once the state transitions to a terminal state. If an implementation specific timeout for the request is reached while the replica waits for the terminal state, then the replica will reply before the call completes, meaning a certificate for the call state in `unknown`, `received`, or `processing` is returned.
+
+The user should take the following actions for the returned certificate states:
+
+-    `processing`: The call is being executed by the IC. The user should poll the IC using [`read_state`](#http-read-state) requests to determine the result of the call.
+
+-    `received`: The call has been received by the IC. The user should poll the IC using [`read_state`](#http-read-state) requests to determine the result of the call.
+
+-    `unknown`: The user can conclude that the message was not received by the IC at the attached timestamp. The user can retry the call, or poll the IC using [`read_state`](#http-read-state) requests.
```

Comment:
> The user can conclude that the message was not received by the IC at the attached timestamp.
This is *wrong*. The message may not be in the certified state tree at that timestamp, but it could be at any later time because the message may indeed be sitting in some artifact pool of some other node and be included in the next block. The message may also have been received earlier and may have been pruned. It's very hard to conclude anything from `unknown` (other than the user may want to poll until expiration to learn whether anything comes up).

Reply:
I'll clarify the `unknown` case. But I still believe it is useful always to return the certificate. If the user/agent sees `processing`, or `received` then they have a signal for their message being received by the IC and that they don't need to retry the request.

Reply:
> they don't need to retry the request.

the request might still time out even after being "received"

Reply:
I agree that in the current setup there is not much value in attaching a certificate for processing, received and unknown.

However, I am at odds with our long term plans. It was said that we should clearly not rely on the ingress history to figure out the if an update made it or not.  So in this case I would imagine that long term, if a user wants to check that his ingress message made it, then the user will have to query in certified fashion the canister state. All this is application specific. 

What is not application specific is that most likely the user will need to be able to distinguish between different timestamps. More specifically the user will need to know past what timestamp querying the canister will yield a correct result.

For example if the user queries the canister at time X to verify that an ingress message mutated the state, then X should be definitely bigger than the time at which a certificate was returned when in processing or received 
 (and unknown?!) state.

All this said, if we omit the certificate for those states now and we want to add them later, most likely it won't break much due to the cbor nature of the response.

Reply:
> What is not application specific is that most likely the user will need to be able to distinguish between different timestamps. More specifically the user will need to know past what timestamp querying the canister will yield a correct result.
Based on the expiry time (which is set by the user), the user can figure out at which point the call must've made it, even without the certificate, though I'm not actually sure where the call will have made it to :) probably the input queue at best, I doubt we guarantee that the message processing actually starts. And I think this would be no better with a certificate. But even if the processing did start at this point, the canister can in principle take an arbitrarily long time to process the message. So the user will in general still need some canister-specific knowledge to figure out at which time the result of a query is definite.

Reply:
I changed it to return 202 and an empty body if it's not in a terminal state. This means we will not return a certificate if the call status is not in `replied`, `rejected`, or `done`.